### PR TITLE
term: remove Dvcfile from repro cmd ref.

### LINKED ISSUE #1445: term: stop using Dvcfile
since there's no longer a default stage name in `dvc run` 1.0

This update was started in https://github.com/iterative/dvc.org/pull/1420/commits/435f4103f031e03d7ca194890c4ec59c59430fd3 — needs to be continued in its own PR.

Docs where the term is mentioned:
- [x] push cmd ref. (#1591)
- [x] repro cmd ref. (#1504)
- ~~install/plugins (#1570)~~
- [x] versioning use case tutorial (#1526)
- Double check any other mentions (the one in install/plugins can stay for legacy compatibility)

### PR DESCRIPTION
**Pull request by sarthakforwet**
The following PR completes one subtask of  #1445 by removing/updating `Dvcfile` from `repro.md`.
This issue has also been referenced under #1255 .
> Also remove `Dvcfile` 

Updations:
- [x] Changed the output of `dvc repro` to the new one.
- [x] Updated the usage of `dvc run` command in Examples.

### CODE REVIEW

**Comment by sarthakforwet** [comment]
@jorgeorpinel  I guess the -c and -R options in `repro.md` also needs to be updated as they rely on specifying a target which is in `dvc.yaml` file now. What's your opinion over that?

**Comment by sarthakforwet** [comment]
@shcheklein  @jorgeorpinel  review this please.  Thanks

**Comment by jorgeorpinel** [comment]
Hi @sarthakforwet this PR does not resolve #1445 completely so please don't use that special term in the description because it would cause #1445 to be closed if this is merged.

File: content/docs/command-reference/repro.md [hunk 1]
```diff
@@ -13,7 +13,7 @@ usage: dvc repro [-h] [-q | -v] [-f] [-s] [-c <path>] [-m] [--dry] [-i]
                  [--no-commit] [--downstream] [targets [targets ...]]
 
 positional arguments:
-  targets        Stage or .dvc file to reproduce. 'Dvcfile' by default.
+  targets        Stage to reproduce.
```
Comment (jorgeorpinel) [inline content/docs/command-reference/repro.md:16]: "or .dvc file" should stay.

File: content/docs/command-reference/repro.md [hunk 2]
```diff
@@ -24,6 +24,8 @@ the dependency graph (a
 by the [stage files](/doc/command-reference/run) (DVC-files with dependencies)
 that are found in the <abbr>project</abbr>. The commands defined in these stages
 can then be executed in the correct order, reproducing pipeline results.
+`dvc repro` relies on the DAG definition that it reads from `dvc.yaml`, and uses
+`dvc.lock` to determine what exactly needs to be run.
```
Comment (jorgeorpinel) [inline content/docs/command-reference/repro.md:28]: Thanks but this is out of scope for this PR. The whole cmd ref. needs to be rewritten so no use in adding a small update here, probably. Unless it was needed to explain the Dvcfile removal but I don' think that's the case?

File: content/docs/command-reference/repro.md [hunk 3]
```diff
@@ -184,17 +182,15 @@ best
...
+$ dvc run -n count -d numbers.txt -d process.py -M count.txt \
            "python process.py numbers.txt > count.txt"
 ```
 
-> Note that using `-f Dvcfile` with `dvc run` above is optional, the stage file
-> name would otherwise default to `count.txt.dvc`. We use `Dvcfile` in this
-> example because that's the default stage file name `dvc repro` will read
-> without having to provide any `targets`.
+> Note that a stage name is required when executing `dvc run`. It can be
+> specified with `-n` (`--name`) option as we did above.
```
Comment (jorgeorpinel) [inline content/docs/command-reference/repro.md:194]: Thanks! In this case I would just get rid of the note altogether.

File: content/docs/command-reference/repro.md [hunk 4]
```diff
@@ -247,16 +243,13 @@ If we now run `dvc repro`, we should see this:
...
-Saving information to 'Dvcfile'.
+Stage 'filter' didn't change, skipping
+Running stage 'count' with command:
+        python3 process.py numbers.txt > count.txt
+Updating lock file 'dvc.lock'
 ```
 
-You can now check that `Dvcfile` and `count.txt` have been updated with the new
+You can now check that `dvc.lock` and `count.txt` have been updated with the new
 information and updated dependency/output file hash values, and a new result,
```
Comment (jorgeorpinel) [inline content/docs/command-reference/repro.md:253]: I'll commit this one:
```suggestion
information: updated dependency/output file hash values, and a new result,
```

File: content/docs/command-reference/repro.md [hunk 5]
```diff
@@ -277,14 +270,13 @@ Now, using the `--downstream` option results in the following output:
 
 ```dvc
 $ dvc repro --downstream
-WARNING: assuming default target 'Dvcfile'.
 Data and pipelines are up to date.
 ```
 
 The reason being that the `text.txt` file is a dependency in the target
```
Comment (jorgeorpinel) [inline content/docs/command-reference/repro.md:276]:
```suggestion
The reason being that the `text.txt` file is a dependency in the `count` stage.
```

File: content/docs/command-reference/repro.md [hunk 6]
```diff
@@ -277,14 +270,13 @@ Now, using the `--downstream` option results in the following output:
...
-[DVC-file](/doc/user-guide/dvc-files-and-directories) (`Dvcfile` by default).
-This `Dvcfile` stage is dependent on `filter.dvc`, which happens first in this
-pipeline (shown in the following figure):
+[DVC-file](/doc/user-guide/dvc-files-and-directories). This `count` stage is
+dependent on `filter` stage, which happens first in this pipeline (shown in the
```
Comment (jorgeorpinel) [inline content/docs/command-reference/repro.md:278]:
```suggestion
This last `count` stage is
dependent on `filter` stage, which happens first in this pipeline (shown in the
```

**Comment by sarthakforwet** [comment]
> this PR does not resolve #1445 completely

@jorgeorpinel  I intend to completely solve that issue here. 
Please can you tell me what about Vim plugin I have to look at?
And apart from `repro.md` I only found the use of `Dvcfile` in `push.md` in  `Example: With dependencies` part.

File: content/docs/command-reference/repro.md (hunk 1 above)
Comment (sarthakforwet) [reply content/docs/command-reference/repro.md:16]: But now the pipeline information is being kept in `dvc.yaml` file and I have seen that no `.dvc` file is being generated when you execute `dvc run` but a new entry is added to `dvc.yaml` file. So how the target would be a .dvc file? 

Rather we can say like -
`Stage to reproduce or one of the entry under dvc.yaml file`.

File: content/docs/command-reference/repro.md (hunk 2 above)
Comment (sarthakforwet) [reply content/docs/command-reference/repro.md:28]: Ok Sure! will add it later when updating command reference.

**Comment by sarthakforwet** [comment]
>  you try to help fully updating the whole dvc repro command reference for 1.0

Yeah sure! would love to do that 😁

**Comment by jorgeorpinel** [comment]
>I intend to completely solve that issue here.

Let's not do this @sarthakforwet. It's a good practice to make PRs small (easy to review), get them merged, and then work on the remaining work in other PRs. I changed the title of this PR to reflect it's scope.

So again, please don't use that special term in the description because it would cause #1445 to be closed. Thanks

**Comment by jorgeorpinel** [comment]
But it would be much more useful if you help entirely updating the `dvc repro` cmd ref. first, TBH.

File: content/docs/command-reference/repro.md (hunk 1 above)
Comment (jorgeorpinel) [reply content/docs/command-reference/repro.md:16]: Nope. That suggestion is redundant. Again, the original text is correct: "Stage or .dvc file to reproduce" but without period `.` BTW.

File: content/docs/command-reference/repro.md (hunk 3 above)
Comment (jorgeorpinel) [reply content/docs/command-reference/repro.md:192]: @sarthakforwet this is not updated, yet you resolved the conversation. Again please don't resolve conversations, I'll do that when I check again. Unresolving. Thanks

File: content/docs/command-reference/repro.md (hunk 1 above)
Comment (sarthakforwet) [reply content/docs/command-reference/repro.md:16]: >  but without period `.`

You mean it should be -  `Stage or DVC file to reproduce` ?

**Comment by sarthakforwet** [comment]
> if you help entirely updating the dvc repro cmd ref. first

Sure, can you guide me with the relevant issues or should I open a new PR dedicated to completely update `repro.md` ?

File: content/docs/command-reference/repro.md [hunk 7]
```diff
@@ -169,7 +165,7 @@ only execute the final stage.
 ## Examples
 
 For simplicity, let's build a pipeline defined below. (If you want get your
-hands-on something more real, see this shot
+hands-on something more real, see this short
```
Comment (jorgeorpinel) [inline content/docs/command-reference/repro.md:168]: Thanks!

**Comment by jorgeorpinel** [comment]
> should I open a new PR dedicated to completely update repro.md

@sarthakforwet yes, this would be useful definitely. But again, no pressure in these any more. I think we've established good collaboration. I recommend you focus on polishing your GSoD application and share it with me for feedback if you haven't. We'll have the rest of July for collaborating on Gtihub.

**Comment by jorgeorpinel** [comment]
Thanks for your contributions BTW!

File: content/docs/command-reference/repro.md (hunk 7 above)
Comment (sarthakforwet) [reply content/docs/command-reference/repro.md:168]: My Pleasure! 😁

**Comment by sarthakforwet** [comment]
>  I recommend you focus on polishing your GSoD application and share it with me for feedback if you haven't.

yeah sure! Thanks!

**Comment by sarthakforwet** [comment]
> Thanks for your contributions BTW!

Pleasure! and thanks for support.

File: content/docs/command-reference/repro.md (hunk 2 above)
Comment (jorgeorpinel) [reply content/docs/command-reference/repro.md]: > See #1572
